frontend/model/state.js

```diff
@@ -329,6 +335,235 @@ const getters = {
+          if (sentPartial) {
+            existingPayment.hash = sentPartial.hash // Q: Why do we need this hash?
+          }
+        }
+
+        payments.push({
+          ...existingPayment,
+          ...p,
+          total: existingAmount,
+          partial: partialAmount > 0,
+          dueIn
+        })
+      }
+      return payments
+    })()
+    const late = (() => {
+      // const currentDistribution = this.groupIncomeAdjustedDistribution
+      // const { pledgeAmount } = this.ourGroupProfile
+      const pMonthstamp = prevMonthstamp(cMonthstamp)
+      const latePayments = []
+      const pastMonth = monthlyPayments[pMonthstamp]
+      // QUESTION: Is this only from the previous month? What if there are payments from 2-3 months ago?
+      if (pastMonth) {
+        const pDate = dateFromMonthstamp(pMonthstamp)
+        const dueIn = lastDayOfMonth(pDate)
+        const adjusted = getters.groupIncomeAdjustedDistributionForMonth(pMonthstamp)
+
+        // This logic below is wrong (based on cypress tests). Commented for later analysis.
+        /*
+        for (const payment of pastMonth.lastAdjustedDistribution) {
+          if (payment.from === this.ourUsername && payment.amount > 0) {
+            // Let A = the amount we owe from the previous distribution.
+            // Let B = the total we've sent to payment.to from the current
+            //         month's paymentsFrom.
+            // Let C = the total amount we "owe" to payment.to from the
+            //         current month's distribution.
+            // Let D = the amount we're pledging this month
+            // Let E = the amount still unpaid for the previous month's distribution,
+            //         calculated as: C > 0 ? A : A + D - B
+            //
+            // If E > 0, then display a row for the late payment.
+            const A = payment.amount
+            const B = this.paymentTotalFromUserToUser(this.ourUsername, payment.to, cMonthstamp)
+            var C = currentDistribution
+              .filter(a => a.from === payment.from && a.to === payment.to)
+            C = C.length > 0 ? C[0].amount : 0
+            const D = pledgeAmount
+            const E = C > 0 ? A : A + D - B
+            if (E > 0) {
+              latePayments.push({
+                username: payment.to,
+                displayName: this.userDisplayName(payment.to),
+                amount: payment.amount, // TODO: include currency (what if it was changed?)
+                // partiaL: TODO add this info as it is in this.paymentsTodo
+                isLate: true,
+                date: dueIn
+              })
+            }
+          }
+        }
+        */
+
+        // On the other hand this seems to work as expected. "partial" is the only thing missing
+        for (const payment of adjusted) {
+          if (payment.from !== ourUsername) {
+            continue
+          }
```

Comment:
Not sure I understand what's going on here...

Reply:
It's hard to explain TBH :/. My suggestion:
- pull the code and uncomment this
- run the tests "group-paying.spec.js" - `npm run cy:open` and see how the late payments are displayed incorrectly in the last test.
- Put the code back as it was: Now the late payments are displayed correctly.

Reply:
It doesn't make much sense to modify the code, because you've already modified it, and so I don't know what modifications to make...

For example, whereas before a received payment contained this information:

```js
                const payment = {
                  hash: paymentHash,
                  data,
                  meta,
                  username: fromUser,
                  displayName: this.userDisplayName(fromUser),
                  date: meta.createdDate,
                  amount: data.amount // TODO: properly display and convert in the correct currency using data.currencyFromTo and data.exchangeRate
                }
                if (this.filterPayment(payment)) {
                  payments.push(payment)
                }
```

Now in this code you're pushing far less information into the payments array:

```js
                const { data, meta } = allPayments[paymentHash]
                payments.push({ hash: paymentHash, data, meta })
```

So I don't know how to modify the line above that reads:

```js
              latePayments.push({
                username: payment.to,
                displayName: this.userDisplayName(payment.to),
                amount: payment.amount, // TODO: include currency (what if it was changed?)
                // partiaL: TODO add this info as it is in this.paymentsTodo
                isLate: true,
                date: dueIn
              })
```

That is something you must do, and then uncomment that code, and comment out the code below.

Reply:
Hum, [here's the change needed](https://user-images.githubusercontent.com/14869087/86582878-443d3a00-bf7a-11ea-9eb6-7cc20d3b828b.png). I've done it in the last commit and pushed it.

Reply:
But of course they will fail, they're completely different data? When uncommented `...payment` isn't pushed.

Reply:
I'm not understanding what you mean. Maybe you should do a talk about this, it would be easier to clarify these stuff and try to move forward this PR